File: ios/Classes/lib/notifications/enumerators/NotificationLifeCycle.swift
public enum NotificationLifeCycle : String, CaseIterable, AbstractEnum {
    
    case Foreground = "Foreground"
    case Background = "Background"
    case AppKilled = "AppKilled"
    
    static func fromString(_ value: String?) -> AbstractEnum? {
        if(value == nil) { return self.allCases[0] }
        return self.allCases.first{ "\($0)" == value }
    }
}

File: packages/awesome_notifications/lib/src/enumerators/notification_life_cycle.dart
/// Application life cycle at new notification change state
enum NotificationLifeCycle {
  Foreground,
  Background,
  Terminated;

  @Deprecated('AppKilled is deprecated, use Terminated instead')
  // ignore: non_constant_identifier_names
  static NotificationLifeCycle get AppKilled =>
      NotificationLifeCycle.Terminated;
}
